Add documentation for the jobs failure policy (#4676)
dapr/docs: daprdocs/content/en/reference/api/jobs_api.md

`dueTime` | An optional time at which the job should be active, or the "one shot" time, if other scheduling type fields are not provided. Accepts a "point in time" string in the format of RFC3339, Go duration string (calculated from creation time), or non-repeating ISO8601.
`repeats` | An optional number of times in which the job should be triggered. If not set, the job runs indefinitely or until expiration.
`ttl` | An optional time to live or expiration of the job. Accepts a "point in time" string in the format of RFC3339, Go duration string (calculated from job creation time), or non-repeating ISO8601.

> `failure_policy` | An optional failure policy for the job. Details of the format are below.

Reviewer: Add to this line what the default value is when unset.

Please can we add a new table with the full `failure_policy` API definitions- add what the defaults are when fields are unset.

Author: Didn't add a table, but different bullet points for the two fields of the `constant` policy. Do you think this is good enough? I find a table might be confusing given that there are two types of policies, and one of them doesn't have any configuration.

> `failure_policy` | An optional failure policy for the job. Details of the format are below. If not set, the job will be retried up to 3 times with a delay of 1 second between retries.

Reviewer 2: ```suggestion
`failure_policy` | An optional failure policy for the job. Details of the format are below. If not set, the job is retried up to 3 times with a delay of 1 second between retries.
```

@daily (or @midnight)  | Run once a day, midnight                   | 0 0 0 * * *
@hourly                | Run once an hour, beginning of hour        | 0 0 * * * *

#### failure_policy

`failure_policy` specifies how the job should handle failures.

It can be set to `constant` or `drop`.

> - The `constant` policy will retry the job up to `max_retries` times, with a delay of `interval` between retries.

Reviewer: Please add that if `max_retries` is not set, it will retry forever.

@daily (or @midnight)  | Run once a day, midnight                   | 0 0 0 * * *
@hourly                | Run once an hour, beginning of hour        | 0 0 * * * *

#### failure_policy

`failure_policy` specifies how the job should handle failures.

It can be set to `constant` or `drop`.

> - The `constant` policy will retry the job based on the configuration

Reviewer 2: ```suggestion
- The `constant` policy retries the job based on the configuration
```

>   - `max_retries` configures how many times the job should be retried. Not setting this will make it retry indefinitely.

Reviewer 2: ```suggestion
  - `max_retries` configures how many times the job should be retried. Not setting this makes it retry indefinitely.
```

@daily (or @midnight)  | Run once a day, midnight                   | 0 0 0 * * *
@hourly                | Run once an hour, beginning of hour        | 0 0 * * * *

#### failure_policy

`failure_policy` specifies how the job should handle failures.

It can be set to `constant` or `drop`.

> - The `constant` policy retries the job based on the configuration.

Reviewer 3: ```suggestion
- The `constant` policy retries the job constantly with the following configuration options.
```

>   - `max_retries` configures how many times the job should be retried. Not setting this makes it retry indefinitely.

Reviewer 3: ```suggestion
  - `max_retries` configures how many times the job should be retried. Defaults to retrying indefinitely.
```

Reviewer 3: Not sure if you can set -1 here or 0? If so them would add something like:

"-1 denotes an unlimited number of retries, while 0 means the request will not be retried."

>   - `interval` configures the delay between retries. Not setting this makes it retry immediately.

Reviewer 3: ```suggestion
  - `interval` configures the delay between retries. Defaults to retrying immediately. Valid values are of the form `200ms`, `15s`, `2m`, etc.
```

Reviewer 3: Not sure if all these are valid but this is what we say for other retry policies: https://docs.dapr.io/operations/resiliency/policies/retries/retries-overview/#spec-metadata
